### ISSUE TITLE
Core/Spells: Death Grip does not match highest in threat

### ISSUE THREAD

**Issue by chaodhib** [TrinityCore/TrinityCore#15403]
The third spell effect of Death Grip, SPELL_EFFECT_ATTACK_ME, isn't working in this particular spell. It should match the caster's threat with the highest in list but it doesn't. 

The third condition ```unitTarget->GetVictim() == m_caster``` at https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/SpellEffects.cpp#L3094 seems to be the cause of the issue.

CORE revision number: 1e71aba2ee656a8407dec63a69d0879e16a070e2

**Comment by Rushor** [comment]
adding an exception like 
unitTarget->GetVictim() == m_caster && m_spellInfo->Id != 49560
could be enough here maybe?

**Comment by tkrokli** [comment]
Sounds interesting. @MrSmite and I are looking for issues like this regarding Death Grip ( #15632 )

I presume the result of your suggested addition would look like this?
```cpp
    if (!unitTarget || !unitTarget->CanHaveThreatList()
        || unitTarget->GetVictim() == m_caster && m_spellInfo->Id != 49560)
```
BTW, about `SendCastResult(SPELL_FAILED_DONT_REPORT);` in https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/SpellEffects.cpp#L3096 -- why does it say `_DONT_REPORT` ?

**Comment by chaodhib** [comment]
Yes, if the dk has already aggro on a unit, if another player has more threat on that unit and if the dk casts death grip, the DK's threat is not modified to match the highest in the list.
